### DISCUSSION TITLE
Why optimizer_zero_grad() is behind backward()

### DISCUSSION THREAD

**Discussion by JulianXiang** [Lightning-AI/pytorch-lightning#6474] [General]
In PL, the whole hook is in https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html?#hooks.
When I see code template about pytorch,  optimizer_zero_grad() is in front of backward(). Why ?

**Comment by akihironitta** [comment]
@AliceNEET Hi, we recently changed the order from `backward`, `step`, `zero_grad` to `zero_grad`,` backward`, `step` in #6147, so if you're using 1.2.2 or later, `zero_grad` is called before `backward`. I'll update the docs in no time to reflect the change.

**Reply by JulianXiang** [reply]
Thanks !!
